src/views/Table.vue

```diff
@@ -0,0 +1,187 @@
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+  <div>
+    <CylcObjectMenu />
+    <div class="c-table">
+      <table-component
+        :tasks="tasks"
+        :hoverable="false"
+        :activable="false"
+        :multiple-active="false"
+        :min-depth="1"
+        ref="table0"
+        key="table0"
+      ></table-component>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mixin } from '@/mixins'
+import { datatable } from '@/mixins/tableview'
+import TableComponent from '@/components/cylc/table/Table'
+import { WORKFLOW_TABLE_DELTAS_SUBSCRIPTION } from '@/graphql/queries'
+import Alert from '@/model/Alert.model'
+import CylcObjectMenu from '@/components/cylc/cylcObject/Menu'
+import { mergeWith } from 'lodash'
+import Vue from 'vue'
+
+/**
+* @param {DeltasAdded} data
+* @param {Array} array
+*/
+const applyTableDeltas = (data, array) => {
+  const added = data.added
+  const pruned = data.pruned
+  const updated = data.updated
+  if (added) {
+    if (added.taskProxies) {
+      for (const taskProxy of added.taskProxies) {
+        array.push(taskProxy)
+      }
+    }
+  }
+  if (pruned) {
+    if (pruned.taskProxies) {
+      for (const taskProxy of pruned.taskProxies) {
+        const indexToRemove = array.findIndex(task => task.id === taskProxy.id)
+        array.splice(indexToRemove, 1)
+      }
+    }
+  }
+  if (updated) {
+    if (updated.taskProxies) {
+      for (const taskProxy of updated.taskProxies) {
+        // const indexToUpdate = array.findIndex(task => task.id === taskProxy)
+        const existingTask = array.find(task => task.id === taskProxy.id)
+        mergeWith(existingTask, taskProxy, mergeWithCustomizer)
+      }
+    }
+  }
+}
+function mergeWithCustomizer (objValue, srcValue, key, object, source) {
+  if (srcValue !== undefined) {
+    // 1. object[key], or objValue, is undefined
+    //    meaning the destination object does not have the property
+    //    so let's add it with reactivity!
+    if (objValue === undefined) {
+      Vue.set(object, `${key}`, srcValue)
+    }
+    // 2. object[key], or objValue, is defined but without reactivity
+    //    this means somehow the object got a new property that is not reactive
+    //    so let's now make it reactive with the new value!
+    if (object[key] && !object[key].__ob__) {
+      Vue.set(object, `${key}`, srcValue)
+    }
+  }
+}
+export default {
+  mixins: [
+    mixin,
+    datatable
+  ],
+
+  name: 'Table',
+
+  props: {
+    workflowName: {
+      type: String,
+      required: true
+    }
+  },
+
+  components: {
+    CylcObjectMenu,
```

Comment:
Nice, this is the component used for the mutations menu. As it's already imported here, I think we have 50% done. Just need to use the directive and confirm it works with the table too.

Reply:
@giulianoserrao has already added the mutations menu :tada: marking as resolved.

Reply:
Directive now being used and working along with Mutations menu